src/Control/Carrier/FossaApiClient/Internal/FossaAPIV1.hs

```diff
@@ -992,13 +1023,14 @@ signedURLEndpoint baseUrl = baseUrl /: "api" /: "components" /: "signed_url"
 getSignedURL ::
   (Has (Lift IO) sig m, Has Debug sig m, Has Diagnostics sig m) =>
   ApiOpts ->
+  ComponentUploadFileType ->
   Text ->
   Text ->
   m SignedURL
-getSignedURL apiOpts revision packageName = fossaReq $ do
+getSignedURL apiOpts fileType revision packageName = fossaReq $ do
   (baseUrl, baseOpts) <- useApiOpts apiOpts
 
-  let opts = "packageSpec" =: packageName <> "revision" =: revision
+  let opts = "packageSpec" =: packageName <> "revision" =: revision <> "fileType" =: toText fileType
```

Comment:
[Optional] I think `toText fileType` would be clearer if there were a function alias like `coreLocatorSpecifier :: ComponentUploadFileType -> Text`. There are a lot of reasons we might want to generate a textual version of that type, in practice `toText` is correct but the name gives a bit of documentation/indirection that might be useful.